server/main_server/src/rodsServer.cpp

```diff
@@ -1099,344 +1113,348 @@ execAgent( int newSock, startupPack_t *startupPack ) {
-                 status );
+int initServer(rsComm_t* svrComm)
+{
+    int status = initServerInfo(0, svrComm);
+    if (status < 0) {
+        rodsLog(LOG_NOTICE, "initServer: initServerInfo error, status = %d", status);
         return status;
     }
 
-    // JMC - legacy resources - printLocalResc ();
     resc_mgr.print_local_resources();
 
     printZoneInfo();
 
-    status = getRcatHost( MASTER_RCAT, NULL, &rodsServerHost );
+    rodsServerHost_t* rodsServerHost{};
+    status = getRcatHost(MASTER_RCAT, nullptr, &rodsServerHost);
 
-    if ( status < 0 || NULL == rodsServerHost ) { // JMC cppcheck - nullptr
+    if (status < 0 || !rodsServerHost) {
         return status;
     }
 
     std::string svc_role;
     irods::error ret = get_catalog_service_role(svc_role);
-    if(!ret.ok()) {
+    if (!ret.ok()) {
         irods::log(PASS(ret));
         return ret.code();
     }
 
 
-    if ( rodsServerHost->localFlag == LOCAL_HOST ) {
-        if( irods::CFG_SERVICE_ROLE_PROVIDER == svc_role ) {
+    if (LOCAL_HOST == rodsServerHost->localFlag) {
+        if (irods::CFG_SERVICE_ROLE_PROVIDER == svc_role) {
             disconnectRcat();
         }
     }
-    else {
-        if ( rodsServerHost->conn != NULL ) {
-            rcDisconnect( rodsServerHost->conn );
-            rodsServerHost->conn = NULL;
-        }
+    else if (rodsServerHost->conn) {
+        rcDisconnect(rodsServerHost->conn);
+        rodsServerHost->conn = nullptr;
     }
 
-    if( irods::CFG_SERVICE_ROLE_PROVIDER == svc_role ) {
-        purgeLockFileDir( 0 );
+    if (irods::CFG_SERVICE_ROLE_PROVIDER == svc_role) {
+        purgeLockFileDir(0);
     }
 
     return status;
 }
 
-/* record the server process number and other information into
-   a well-known file.  If svrComm is Null and this has created a file
-   before, just unlink the file. */
-int
-recordServerProcess( rsComm_t *svrComm ) {
-    int myPid;
-    FILE *fd;
-    DIR  *dirp;
+// record the server process number and other information into
+// a well-known file. If svrComm is Null and this has created a file
+// before, just unlink the file.
+int recordServerProcess(rsComm_t* svrComm)
+{
     static char filePath[100] = "";
-    char cwd[1000];
-    char stateFile[] = "irodsServer";
-    char *tmp;
-    char *cp;
 
-    if ( svrComm == NULL ) {
-        if ( filePath[0] != '\0' ) {
-            unlink( filePath );
+    if (!svrComm) {
+        if (filePath[0] != '\0') {
+            unlink(filePath);
         }
+
         return 0;
     }
-    rodsEnv *myEnv = &( svrComm->myEnv );
 
-    /* Use /usr/tmp if it exists, /tmp otherwise */
-    dirp = opendir( "/usr/tmp" );
-    if ( dirp != NULL ) {
+    rodsEnv* myEnv = &svrComm->myEnv;
+
+    // Use /usr/tmp if it exists, /tmp otherwise.
+    char* tmp;
+    if (DIR* dirp = opendir("/usr/tmp"); dirp) {
         tmp = "/usr/tmp";
-        ( void )closedir( dirp );
+        (void) closedir(dirp);
```

Comment:
Is this masking a return code? Should we check it? Or use a C++-style cast?

Reply:
My guess is that it is keeping the compiler from screaming about something.

I'll see what happens if we remove `void`.

Reply:
I believe the cast isn't necessary anymore.